lib/cocoapods-core/http.rb

```diff
@@ -59,17 +59,19 @@ def self.validate_url(url)
     #
     # @return [REST::response]
     #
-    def self.perform_head_request(url)
+    def self.perform_head_request(url, user_agent)
       require 'rest'
 
-      resp = ::REST.head(url, 'User-Agent' => USER_AGENT)
+      user_agent ||= USER_AGENT
```

Comment:
can also set that as the default?
```ruby
def self.perform_head_request(url, user_agent = USER_AGENT)
...
end
```

Reply:
I did at first, but realized this method is private to http.rb, so it becomes easier to pass nil at the call site(s) and then substitute it here.